tests/components/maxcube/test_maxcube_climate.py

```diff
@@ -0,0 +1,273 @@
+"""Test EQ3 Max! Thermostats."""
+from conftest import updated_state
+from maxcube.cube import MaxCube
+from maxcube.device import (
+    MAX_DEVICE_MODE_AUTOMATIC,
+    MAX_DEVICE_MODE_BOOST,
+    MAX_DEVICE_MODE_MANUAL,
+    MAX_DEVICE_MODE_VACATION,
+)
+from maxcube.thermostat import MaxThermostat
+from maxcube.wallthermostat import MaxWallThermostat
+
+from homeassistant.components.climate.const import (
+    ATTR_CURRENT_TEMPERATURE,
+    ATTR_HVAC_ACTION,
+    ATTR_HVAC_MODE,
+    ATTR_HVAC_MODES,
+    ATTR_MAX_TEMP,
+    ATTR_MIN_TEMP,
+    ATTR_PRESET_MODE,
+    ATTR_PRESET_MODES,
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    CURRENT_HVAC_OFF,
+    DOMAIN as CLIMATE_DOMAIN,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    PRESET_AWAY,
+    PRESET_BOOST,
+    PRESET_COMFORT,
+    PRESET_ECO,
+    PRESET_NONE,
+    SERVICE_SET_HVAC_MODE,
+    SERVICE_SET_PRESET_MODE,
+    SERVICE_SET_TEMPERATURE,
+)
+from homeassistant.components.maxcube.climate import (
+    MAX_TEMPERATURE,
+    MIN_TEMPERATURE,
+    OFF_TEMPERATURE,
+    ON_TEMPERATURE,
+    PRESET_ON,
+    SUPPORT_FLAGS,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_FRIENDLY_NAME,
+    ATTR_SUPPORTED_FEATURES,
+    ATTR_TEMPERATURE,
+)
+
+ENTITY_ID = "climate.testroom_testthermostat"
+WALL_ENTITY_ID = "climate.testroom_testwallthermostat"
+VALVE_POSITION = "valve_position"
+
+
+def _set_hvac_mode(hass, entity_id: str, mode: str):
+    hass.services.call(
```

Comment:
We should use the async api in the tests. All the tests should be coroutine functions.

Reply:
Replaced by await hass.servicies.async_call(...)